File: llm_generated_code/Gemini/CPP/Level2/Q1.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<int> chocolates = {5, 2, 12, 20, 1};
    int max_chocolates = chocolates[0];

    // Time complexity: O(n)
    // Space complexity: O(1)
    // Lines of code: 6
    // Cyclomatic complexity: 2
    for (int i = 1; i < chocolates.size(); i++) {
        if (chocolates[i] > max_chocolates) {
            max_chocolates = chocolates[i];
        }
    }

    cout << "The highest number of chocolates is: " << max_chocolates << endl;

    return 0;
}

File: llm_generated_code/Gemini/Python/Level2/Q1.py
def find_max_chocolates(chocolates):
    """
    Finds the maximum number of chocolates among a list of friends.

    Args:
        chocolates: A list of integers representing the number of chocolates each friend has.

    Returns:
        The maximum number of chocolates.
    """

    max_chocolates = 0
    for chocolate in chocolates:
        if chocolate > max_chocolates:
            max_chocolates = chocolate
    return max_chocolates

# Example usage:
chocolates = [5, 2, 12, 20, 1]
max_chocolates = find_max_chocolates(chocolates)
print("The highest number of chocolates is:", max_chocolates)
